src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -6034,4 +6134,117 @@ export class StudyViewPageStore {
         }
         return title;
     }
+
+    @observable
+    private _pTreatmentsFilter: AndedPatientTreatmentFilters = { filters: [] };
+
+    @observable
+    private _sTreatmentsFilters: AndedSampleTreatmentFilters = { filters: [] };
+
+    @computed
+    public get patientTreatmentFilters(): AndedPatientTreatmentFilters {
+        return this._pTreatmentsFilter;
+    }
+
+    @computed
+    public get sampleTreatmentFilters(): AndedSampleTreatmentFilters {
+        return this._sTreatmentsFilters;
+    }
+
+    @action
+    public clearPatientTreatmentFilters() {
+        this._pTreatmentsFilter = { filters: [] };
+    }
+
+    @action
+    public setPatientTreatmentFilters(filters: AndedPatientTreatmentFilters) {
+        this._pTreatmentsFilter = filters;
+    }
+
+    @action
+    public addPatientTreatmentFilters(filters: OredPatientTreatmentFilters[]) {
+        this._pTreatmentsFilter.filters = this._pTreatmentsFilter.filters.concat(
+            filters
+        );
+    }
+
+    @action
+    public clearSampleTreatmentFilters() {
+        this._sTreatmentsFilters = { filters: [] };
+    }
+
+    @action
+    public setSampleTreatmentFilters(filters: AndedSampleTreatmentFilters) {
+        this._sTreatmentsFilters = filters;
+    }
+
+    @action
+    public addSampleTreatmentFilters(filters: OredSampleTreatmentFilters[]) {
+        this._sTreatmentsFilters.filters = this._sTreatmentsFilters.filters.concat(
+            filters
+        );
+    }
+
+    @action
+    public setBothFilters(filters: Partial<StudyViewFilter>) {
+        if (
+            filters.patientTreatmentFilters &&
+            _.isArray(filters.patientTreatmentFilters.filters)
+        ) {
+            this.setPatientTreatmentFilters(filters.patientTreatmentFilters);
+        }
+        if (
+            filters.sampleTreatmentFilters &&
+            _.isArray(filters.sampleTreatmentFilters.filters)
+        ) {
+            this.setSampleTreatmentFilters(filters.sampleTreatmentFilters);
+        }
+    }
+
+    // a row represents a list of patients that either have or have not recieved
+    // a specific treatment
+    @computed
+    public get sampleTreatments() {
```

Comment:
any specific reason to get create this promise with computed. why not create it directly